src/neural/blas/convolution1.h

```diff
@@ -0,0 +1,37 @@
+/*
+ This file is part of Leela Chess Zero.
+ Copyright (C) 2018 The LCZero Authors
+
+ Leela Chess is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Leela Chess is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <vector>
+
+namespace lczero {
+
+class Convolution1 {
+ public:
+  static void Forward(const size_t batch_size, const size_t input_channels,
```

Comment:
Comment to a public function. Also comment to a class would be useful (why it is Convolution1 and not simply Convolution?)

Reply:
1 is the spacial parameter. It's a 1x1 convolution.  I will make this comment:

```
// Convolution 1x1
class Convolution1 {
  public:
  static void Forward(const size_t batch_size, const size_t input_channels,
```